web/modules/custom/nys_metatags/nys_metatags.module

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/**
+ * @file
+ * Contains custom code for Metatags.
+ */
+
+use Drupal\file\Entity\File;
+use Drupal\node\NodeInterface;
+
+/**
+ * Alter the meta tags for any page prior to page attachment.
+ *
+ * @param array $metatag_attachments
+ *   An array of metatag objects to be attached to the current page.
+ */
+function nys_metatags_metatags_attachments_alter(array &$metatag_attachments) {
+  $node = \Drupal::routeMatch()->getParameter('node');
+  if (!($node instanceof NodeInterface)) {
+    return;
+  }
+  if ($node->bundle() === 'honoree') {
+    $title = $node->title->value ?? '';
+    if (!$title) {
+      return;
+    }
+    $type = $node->field_honoree_type[0]->value;
+    foreach ($metatag_attachments['#attached']['html_head'] as $id => $attachment) {
+      if ($attachment[1] == 'title') {
+        if ($type === 'women') {
+          $metatag_attachments['#attached']['html_head'][$id][0]['#attributes']['content'] = $title . ' | Women of Distinction | NYSenate.gov';
+        }
+        elseif ($type === 'veteran') {
+          $metatag_attachments['#attached']['html_head'][$id][0]['#attributes']['content'] = $title . ' | Veteran\'s Hall of Fame | NYSenate.gov';
+        }
+      }
+      if ($attachment[1] == 'og_title' || $attachment[1] == 'twitter_cards_title') {
+        if ($type === 'women') {
+          $metatag_attachments['#attached']['html_head'][$id][0]['#attributes']['content'] = $title . ' | Women of Distinction';
+        }
+        elseif ($type === 'veteran') {
+          $metatag_attachments['#attached']['html_head'][$id][0]['#attributes']['content'] = $title . ' | Veteran\'s Hall of Fame';
+        }
+      }
+    }
+  }
+  if ($node->bundle() === 'microsite_page') {
+    $title = $node->title->value ?? '';
+    if (!$title) {
+      return;
+    }
+    $type = $node->field_microsite_page_type[0]->target_id;
+    $senator_id = $node->field_senator_multiref[0]->target_id;
+    try {
+      $senator = \Drupal::entityTypeManager()->getStorage('taxonomy_term')
+        ->load($senator_id);
+    }
+    catch (\Throwable) {
+      $senator = NULL;
+    }
+
+    if (isset($senator->name->value) && !empty($senator->name->value)) {
```

Comment:
If the `if()` statement fails, `$senator_name` will be uninitialized.  Use this instead:

`$senator_name = $senator->name->value ?? ''`